src/Util/Test.php

```diff
@@ -517,7 +518,7 @@ private static function getDataFromDataProviderAnnotation($docComment, $classNam
                     $data = $dataProviderMethod->invoke($object, $methodName);
                 }
 
-                if ($data instanceof Iterator) {
+                if ($data instanceof Traversable) {
                     $data = \iterator_to_array($data);
```

Comment:
I referred to main change of this PR.
And my proposal is deeper - to allow dataProvider not only work properly with Traversable, but all iterable.

whith this change, it will probably be like that, good to prove by utests, which are already there